virttest/utils_test/__init__.py

```diff
@@ -1947,149 +1956,147 @@ def _tarball_download(self, url, destination):
         self.base_name = tarball.members[0].name.split("/")[0]
         tarball.close()
 
-    @error_context.context_aware
     def download_stress(self):
         """
         Download stress tool
         """
         url = self.download_url
         tmp_path = data_dir.get_tmp_dir()
-        error_context.context('Download stress tool from %s' % url,
-                              logging.info)
+        logging.info('Download stress tool from %s', url)
         download_method = getattr(self, "_%s_download" % self.download_type)
         download_method(url, tmp_path)
-        error_context.context('Copy stress tool to guest', logging.info)
-        self.vm.wait_for_login()
-        self.vm.copy_files_to(os.path.join(tmp_path, self.base_name), self.dst_path)
+        source = os.path.join(tmp_path, self.base_name)
+        if self.remote_host:
+            logging.info('Copy stress tool to remote host')
+
+            args = (self.remote_host.__getitem__('server_ip'), 'scp',
+                    self.remote_host.__getitem__('server_user'),
+                    self.remote_host.__getitem__('server_pwd'), "22",
+                    source, self.dst_path)
+            self.copy_files_to(args)
+        else:
+            logging.info('Copy stress tool to work dir of host/guest')
+            self.copy_files_to(source, self.dst_path)
```

Comment:
For vm, it probably fail if not make sure guest is accessible (by `vm.wait_for_login()` or maybe `vm.verify_alive()`) before `copy_file_to`

Reply:
@vivianQizhu you are right, but while entering `load_stress_tool()`, @session_handler would raise StressError if `vm.wait_for_login()` fails. This could ensure the guest is accessible.